awscli/topictags.py

```diff
@@ -0,0 +1,254 @@
+            os.path.abspath(__file__)), 'topics')
+
+    # The default JSON index to load.
+    JSON_INDEX = os.path.join(TOPIC_DIR, 'topic-tags.json')
+
+    def __init__(self, tag_dictionary=None):
+        self._tag_dictionary = tag_dictionary
+        if self._tag_dictionary is None:
+            self._tag_dictionary = {}
+
+    def load_json_index(self, index_file=None):
+        """Loads a JSON file into the tag dictionary.
+
+        :param index_file: The path to a specific JSON index to load.
+            If nothing is specified it will default to the default JSON
+            index at ``JSON_INDEX``.
+        """
+        index_filepath = self.JSON_INDEX
+        if index_file is not None:
+            index_filepath = index_file
+        with open(index_filepath, 'r') as f:
+            index = f.read()
+            self._tag_dictionary = json.loads(index)
+
+    def save_to_json_index(self, index_file=None):
+        """Writes the loaded data back out to the JSON index.
+
+        :param index_file: The path to a specific JSON index to load.
+            If nothing is specified it will default to the the default
+            JSON index at ``JSON_INDEX``.
+        """
+        index_filepath = self.JSON_INDEX
+        if index_file is not None:
+            index_filepath = index_file
+        with open(index_filepath, 'w') as f:
+            f.write(json.dumps(self._tag_dictionary, indent=4, sort_keys=True))
+
+    def get_all_topic_names(self):
+        """Retrieves all of the topic names of the loaded JSON index"""
+        return list(self._tag_dictionary.keys())
+
+    def get_all_topic_src_files(self):
+        """Retrieves the file paths of all the topics in directory"""
+        topic_full_paths = []
+        topic_names = os.listdir(self.TOPIC_DIR)
+        for topic_name in topic_names:
+            # Do not try to load hidden files.
+            if not topic_name.startswith('.'):
+                topic_full_path = os.path.join(self.TOPIC_DIR, topic_name)
+                # Ignore the JSON Index as it is stored with topic files.
+                if topic_full_path != self.JSON_INDEX:
+                    topic_full_paths.append(topic_full_path)
+        return topic_full_paths
+
+    def scan(self, topic_files):
+        """Scan in the tags of a list of topics into memory.
+
+        Note that if there are existing values in an entry in the database
+        of tags, they will not be overwritten. Any new values will be
+        appended to original values.
+
+        :param topic_files: A list of paths to topics to scan into memory.
+        """
+        for topic_file in topic_files:
+            with open(topic_file, 'r') as f:
+                # Parse out the name of the topic
+                topic_name = self._find_topic_name(topic_file)
+                # Add the topic to the dictionary if it does not exist
+                self._add_topic_name_to_dict(topic_name)
+                # Read the file
```

Comment:
FWIW, this comment seems unnecessary, it's pretty clear that `f.read()` reads the contents of a file.